Game/Scripts/DroneExplosionAttack.cpp

```diff
@@ -0,0 +1,100 @@
+#include "DroneExplosionAttack.h"
+
+#include "Components/Component.h"
+#include "Components/ComponentScript.h"
+#include "Components/ComponentRigidBody.h"
+#include "Components/ComponentTransform.h"
+#include "Components/ComponentAudioSource.h"
+
+#include "../Scripts/HealthSystem.h"
+#include "../Scripts/EnemyDroneScript.h"
+
+#include "Auxiliar/Audio/AudioData.h"
+
+REGISTERCLASS(DroneExplosionAttack);
+
+DroneExplosionAttack::DroneExplosionAttack() : Script(),
+	attackState(DroneExplosionState::NOTDEAD), targetPlayer(nullptr), explosionDamage(30.0f),
+	explosionTime(3.0f)
+{
+	REGISTER_FIELD(explosionDamage, float);
+	REGISTER_FIELD(explosionTime, float);
+}
+
+void DroneExplosionAttack::Start()
+{
+	rigidBody = owner->GetComponent<ComponentRigidBody>();
+	parentTransform = owner->GetParent()->GetComponent<ComponentTransform>();
+	parentHealthSystem = owner->GetParent()->GetComponent<HealthSystem>();
+	parentEnemyDroneScript = owner->GetParent()->GetComponent<EnemyDroneScript>();
+	componentAudioSource = owner->GetParent()->GetComponent<ComponentAudioSource>();
+	rigidBody->SetKpForce(50);
+}
+
+void DroneExplosionAttack::Update(float deltaTime)
+{
+
+
+	if (parentHealthSystem->GetField<float>("CurrentHealth")->getter() <= 10.0f 
```

Comment:
```suggestion
	if (parentHealthSystem->GetCurrentHealth() <= 10.0f 
```
Try not to use `GetField` to get the variables from other scripts, we shouldn't be using that. Instead, we could use a basic getter (HealthSystem script has `GetCurrentHealth()`, so you can use that) or make the variables public.

Reply:
Done